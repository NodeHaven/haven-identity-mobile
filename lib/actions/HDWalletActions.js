// Copyright (C) 2018 ConsenSys AG
//
// This file is part of uPort Mobile App.
//
// uPort Mobile App is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// uPort Mobile App is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with uPort Mobile App.  If not, see <http://www.gnu.org/licenses/>.
//
import { STORE_ROOT_ADDRESS, INC_IDENTITY_INDEX, INC_ACCOUNT_INDEX, SEED_CONFIRMED, ACCOUNT_RISK_SENT, RESET_HD_WALLET, STORE_ALL_KEYCHAIN_ADDRESS } from 'uPortMobile/lib/constants/HDWalletActionTypes'

// These are used to keep a track of used and unused paths for a HD wallet

// ADDING NEW ACTIONS PLEASE READ THIS!!!!
//
// If the action should be backed up add the `, _backup` attribute to the action.
// When should it be backed up?
//
// - is it backed by the reducer and not a saga?
// - is it something that can not be refreshed from the block chain etc? (eg. no need to backup nonces and balances)
// - is it not just related to UX? Some reducers are temporary and would not make sense on multiple device

export function storeRootAddress (root) {
  return {
    type: STORE_ROOT_ADDRESS,
    root,
    _backup: true
  }
}

export function incIdentityIndex () {
  return {
    type: INC_IDENTITY_INDEX,
    _backup: true
  }
}

export function incAccountIndex (identity) {
  return {
    type: INC_ACCOUNT_INDEX,
    identity,
    _backup: true
  }
}

export function seedConfirmed () {
  return {
    type: SEED_CONFIRMED,
    _backup: true
  }
}

export function accountRiskSent () {
  return {
    type: ACCOUNT_RISK_SENT,
    _backup: true
  }
}

export function resetHDWallet () {
  return {
    type: RESET_HD_WALLET
  }
}


export function storeAllKeyChainAddresses (addresses) {
  return {
    type: STORE_ALL_KEYCHAIN_ADDRESS,
    addresses
  }
}