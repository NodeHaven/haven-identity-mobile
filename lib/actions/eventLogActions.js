// Copyright (C) 2018 ConsenSys AG
//
// This file is part of uPort Mobile App.
//
// uPort Mobile App is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// uPort Mobile App is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with uPort Mobile App.  If not, see <http://www.gnu.org/licenses/>.
//
import { STORE_LAST_BLOCK_FOR_LOGS, STORE_EVENTS } from '../constants/EventLogActionTypes'

// ADDING NEW ACTIONS PLEASE READ THIS!!!!
//
// If the action should be backed up add the `, _backup` attribute to the action.
// When should it be backed up?
//
// - is it backed by the reducer and not a saga?
// - is it something that can not be refreshed from the block chain etc? (eg. no need to backup nonces and balances)
// - is it not just related to UX? Some reducers are temporary and would not make sense on multiple device

const _backup = true

export function storeLastBlockForLogs (contract, block) {
  return {
    type: STORE_LAST_BLOCK_FOR_LOGS,
    contract,
    block,
    _backup
  }
}

export function storeEvents (contract, events) {
  return {
    type: STORE_EVENTS,
    contract,
    events,
    _backup
  }
}
