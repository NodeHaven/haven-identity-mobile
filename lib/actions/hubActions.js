// Copyright (C) 2018 ConsenSys AG
//
// This file is part of uPort Mobile App.
//
// uPort Mobile App is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// uPort Mobile App is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with uPort Mobile App.  If not, see <http://www.gnu.org/licenses/>.
//
import { STORE_HUB_HEAD, CATCHUP_HUB, QUEUE_EVENT, TAKE_QUEUED_EVENT, CLEAR_QUEUE, IMPORT_SNAPSHOT, RESET_HUB } from '../constants/HubActionTypes'

export function storeHubHead (hash) {
  return {
    type: STORE_HUB_HEAD,
    hash
  }
}
export function resetHub () {
  return {
    type: RESET_HUB
  }
}

export function catchupHub () {
  return {
    type: CATCHUP_HUB
  }
}

export function queueEvent (event) {
  return {
    type: QUEUE_EVENT,
    event
  }
}

export function takeQueuedEvent () {
  return {
    type: TAKE_QUEUED_EVENT
  }
}

export function clearQueue () {
  return {
    type: CLEAR_QUEUE
  }
}

export function importSnapshot (snapshot) {
  return {
    type: IMPORT_SNAPSHOT,
    snapshot
  }
}
