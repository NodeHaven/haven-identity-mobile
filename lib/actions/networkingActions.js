// Copyright (C) 2018 ConsenSys AG
//
// This file is part of uPort Mobile App.
//
// uPort Mobile App is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// uPort Mobile App is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with uPort Mobile App.  If not, see <http://www.gnu.org/licenses/>.
//
import {
  ANALYZE_ADDRESS,
  MARK_OFFLINE,
  MARK_ONLINE,
  SAVE_GAS_PRICE
} from 'uPortMobile/lib/constants/NetworkingActionTypes'

export function goOffline () {
  return {
    type: MARK_OFFLINE
  }
}

export function goOnline () {
  return {
    type: MARK_ONLINE
  }
}

export function saveGasPrice (gasPrice) {
  return {
    type: SAVE_GAS_PRICE,
    gasPrice
  }
}

export function analyzeAddress (address) {
  return {
    type: ANALYZE_ADDRESS,
    address
  }
}
