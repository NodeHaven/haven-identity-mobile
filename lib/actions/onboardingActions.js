// Copyright (C) 2018 ConsenSys AG
//
// This file is part of uPort Mobile App.
//
// uPort Mobile App is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// uPort Mobile App is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with uPort Mobile App.  If not, see <http://www.gnu.org/licenses/>.
//
import { ADD_DATA, FETCH_CURRENT_COUNTRY, FLAG_NEWBIE, UNFLAG_NEWBIE, SET_ONBOARDING_NETWORK } from '../constants/OnboardingActionTypes'

export function addData (data) {
  return {
    type: ADD_DATA,
    data
  }
}

export function fetchCurrentCountry () {
  return {
    type: FETCH_CURRENT_COUNTRY
  }
}

export function flagNewbie () {
  return {
    type: FLAG_NEWBIE
  }
}

export function unflagNewbie () {
  return {
    type: UNFLAG_NEWBIE
  }
}

export function setOnboardingNetwork (network) {
  return {
    type: SET_ONBOARDING_NETWORK,
    network
  }
}
