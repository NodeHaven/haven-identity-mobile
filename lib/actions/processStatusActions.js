// Copyright (C) 2018 ConsenSys AG
//
// This file is part of uPort Mobile App.
//
// uPort Mobile App is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// uPort Mobile App is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with uPort Mobile App.  If not, see <http://www.gnu.org/licenses/>.
//
import {
  SAVE_MESSAGE,
  SAVE_ERROR,
  CLEAR_MESSAGE,
  CLEAR_ALL_MESSAGES,
  START_WORKING,
  STOP_WORKING,
  COMPLETE_PROCESS,
  FAIL_PROCESS
} from 'uPortMobile/lib/constants/ProcessStatusActionTypes'

export const saveMessage = (section, message) => ({type: SAVE_MESSAGE, section, message})
export const saveError = (section, error) => ({type: SAVE_ERROR, section, error})
export const clearMessage = (section) => ({type: CLEAR_MESSAGE, section})
export const clearAllMessages = () => ({type: CLEAR_ALL_MESSAGES})
export const startWorking = (section) => ({type: START_WORKING, section})
export const stopWorking = (section) => ({type: STOP_WORKING, section})
export const completeProcess = (section) => ({type: COMPLETE_PROCESS, section})
export const failProcess = (section, error) => ({type: FAIL_PROCESS, section, error})
