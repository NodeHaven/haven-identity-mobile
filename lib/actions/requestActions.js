// Copyright (C) 2018 ConsenSys AG
//
// This file is part of uPort Mobile App.
//
// uPort Mobile App is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// uPort Mobile App is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with uPort Mobile App.  If not, see <http://www.gnu.org/licenses/>.
//
import {
  HANDLE_URL,
  HANDLE_MESSAGE,
  SELECT_REQUEST,
  SELECT_ACCOUNT_FOR_REQUEST,
  SAVE_REQUEST,
  CLEAR_REQUEST,
  AUTHORIZE_REQUEST,
  CANCEL_REQUEST,
  CALLBACK,
  AUTHORIZE_ACCOUNT
} from 'uPortMobile/lib/constants/RequestActionTypes'

export function handleURL (url, options = {}) {
  return {
    type: HANDLE_URL,
    url,
    ...{postback: false, popup: true, clientId: null},
    ...options
  }
}

export function handleMessage (message, options = {}) {
  return {
    type: HANDLE_MESSAGE,
    message,
    ...options
  }
}

export function selectRequest (activityId) {
  return {
    type: SELECT_REQUEST,
    activityId
  }
}

// Use this in a request dialog to select the correct account
export function selectAccountForRequest (activityId, account) {
  return {
    type: SELECT_ACCOUNT_FOR_REQUEST,
    activityId,
    account
  }
}

export function saveRequest (activityId) {
  return {
    type: SAVE_REQUEST,
    activityId
  }
}

export function clearRequest () {
  return {
    type: CLEAR_REQUEST
  }
}

export function authorizeRequest (activityId) {
  return {
    type: AUTHORIZE_REQUEST,
    activityId
  }
}

export function authorizeAccount (activityId, accType) {
  return {
    type: AUTHORIZE_ACCOUNT,
    activityId: activityId,
    accType: accType
  }
}

export function cancelRequest (activityId) {
  return {
    type: CANCEL_REQUEST,
    activityId
  }
}

export function callback (request) {
  return {
    type: CALLBACK,
    request
  }
}
