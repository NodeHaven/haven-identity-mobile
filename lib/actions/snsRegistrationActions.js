// Copyright (C) 2018 ConsenSys AG
//
// This file is part of uPort Mobile App.
//
// uPort Mobile App is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// uPort Mobile App is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with uPort Mobile App.  If not, see <http://www.gnu.org/licenses/>.
//
import {
  REGISTER_DEVICE_FOR_NOTIFICATIONS,
  RECEIVE_DEVICE_TOKEN,
  STORE_ENDPOINT_ARN,
  STORE_DEVICE_TOKEN,
  SKIP_PUSH_NOTIFICATIONS,
  SEND_LOCAL_NOTIFICATION,
  CANCEL_SCHEDULED_NOTIFICATION,
  UPDATE_BADGE_COUNT,
  HANDLE_ENCRYPTED_MESSAGE,
  CHECK_FOR_NOTIFICATIONS,
  POLL_FOR_NOTIFICATIONS,
  INIT_NOTIFICATIONS
} from 'uPortMobile/lib/constants/SnsRegistrationActionTypes'

export function registerDeviceForNotifications () {
  return {
    type: REGISTER_DEVICE_FOR_NOTIFICATIONS
  }
}

export function skipPushNotifications () {
  return {
    type: SKIP_PUSH_NOTIFICATIONS
  }
}

export function storeEndpointArn (endpointArn) {
  return {
    type: STORE_ENDPOINT_ARN,
    endpointArn
  }
}

export function receiveDeviceToken (deviceToken) {
  return {
    type: RECEIVE_DEVICE_TOKEN,
    deviceToken
  }
}

export function storeDeviceToken (deviceToken) {
  return {
    type: STORE_DEVICE_TOKEN,
    deviceToken
  }
}

export function sendLocalNotification (activityId, message, repeatInterval) {
  return {
    type: SEND_LOCAL_NOTIFICATION,
    message,
    activityId,
    repeatInterval
  }
}

export function cancelScheduledNotification (activityId) {
  return {
    type: CANCEL_SCHEDULED_NOTIFICATION,
    activityId
  }
}

export function updateBadgeCount () {
  return {
    type: UPDATE_BADGE_COUNT
  }
}

export function checkForNotifications () {
  return {
    type: CHECK_FOR_NOTIFICATIONS
  }
}

export function pollForNotifications () {
  return {
    type: POLL_FOR_NOTIFICATIONS
  }
}

export function handleEncryptedMessage (id) {
  return {
    type: HANDLE_ENCRYPTED_MESSAGE,
    id
  }
}

export function initNotifications () {
  return {
    type: INIT_NOTIFICATIONS
  }
}
