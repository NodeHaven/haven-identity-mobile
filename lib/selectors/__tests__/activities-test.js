// Copyright (C) 2018 ConsenSys AG
//
// This file is part of uPort Mobile App.
//
// uPort Mobile App is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// uPort Mobile App is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with uPort Mobile App.  If not, see <http://www.gnu.org/licenses/>.
//
import uportReducer from 'uPortMobile/lib/reducers/uportReducer'
import { 
  unread, 
  unreadCount, 
  history, 
  historyCount, 
  activities, 
  notifications,
  notificationCount,
  activitiesForAddress, 
  activitiesByAddress, 
  unconfirmedTransactions 
} from 'uPortMobile/lib/selectors/activities'
import { toJs, toClj, assocIn, vector } from 'mori'
import { massage } from 'uPortMobile/lib/sagas/stateSaver'
import { storeIdentity } from 'uPortMobile/lib/actions/uportActions'

const uport = massage(toClj({
  byAddress: {
    '0x9df0e9759b17f34e9123adbe6d3f25d54b72ad6a': {
      address: '0x9df0e9759b17f34e9123adbe6d3f25d54b72ad6a',
      activities: {
        '10': {
          id: 10,
          type: 'sign',
          authorizedAt: 14834004807621990,
          client_id: '0x1bc5cbf71b068642fc5e89c5e3d0bdb3e366bc01',
          to: '0x1bc5cbf71b068642fc5e89c5e3d0bdb3e366bc02',
        },
        '11': {
          id: 11,
          type: 'connect',
          authorizedAt: 14834004807621990,
          to: '0x1bc5cbf71b068642fc5e89c5e3d0bdb3e366bc02',
        },
        '12': {
          id: 12,
          type: 'disclosure',
          client_id: '0x1bc5cbf71b068642fc5e89c5e3d0bdb3e366bc03'
        },
        '13': {
          id: 13,
          type: 'attestation',
          iss: '0x1bc5cbf71b068642fc5e89c5e3d0bdb3e366bc02'
        },
        '18': {
          id: '18',
          type: 'attestation',
          authorizedAt: 14834004807621990,
          opened: true,
          attestations: [{name: 'Bob'}],
        },
        '19': {
          id: '19',
          type: 'attestation',
          authorizedAt: 14834004807621990,
          attestations: [{name: 'Bob'}],
        }
      },
      stats: {
        '0x1bc5cbf71b068642fc5e89c5e3d0bdb3e366bc01': {
          request: 3,
          sign: 1
        }
      }
    }
  },
  currentIdentity: '0x9df0e9759b17f34e9123adbe6d3f25d54b72ad6a',
  external: {
    '0x1bc5cbf71b068642fc5e89c5e3d0bdb3e366bc01': {
      address: '0x1bc5cbf71b068642fc5e89c5e3d0bdb3e366bc01',
      name: 'Coinbase'
    },
    '0x1bc5cbf71b068642fc5e89c5e3d0bdb3e366bc02': {
      address: '0x1bc5cbf71b068642fc5e89c5e3d0bdb3e366bc02',
      name: 'Gnosis'
    }
  }
}))

const withIdentity = {uport}

const otherIdentity = {
  network: 'kovan',
  address: '34ukSmiK1oA1C5Du8aWpkjFGALoH7nsHeDX',
  own: {
    name: 'Alice Kovan',
    phone: '555-555-5555',
  },
  activities: {
    '14': {
      id: '14',
      type: 'sign',
      authorizedAt: 14834004807621990,
      client_id: '0x1bc5cbf71b068642fc5e89c5e3d0bdb3e366bc01',
      to: '0x1bc5cbf71b068642fc5e89c5e3d0bdb3e366bc02',
    },
    '15': {
      id: '15',
      type: 'connect',
      authorizedAt: 14834004807621990,
      to: '0x1bc5cbf71b068642fc5e89c5e3d0bdb3e366bc02',
    }
  }
}
 
const withMultipleIdentities = {
  uport: uportReducer(withIdentity.uport, storeIdentity(otherIdentity))
}

const identityWithTransactions = {
  network: 'ropsten',
  address: '34ukSmiK1oA1C5Du8aWpkjFGALoH7nsHeDX',
  own: {
    name: 'Ropsten User',
    phone: '555-555-5555',
  },
  activities: {
    '16': {
      id: '14',
      type: 'sign',
      txhash: 'abcd',
      blockNumber: 1231,
      authorizedAt: 14834004807621990,
      client_id: '0x1bc5cbf71b068642fc5e89c5e3d0bdb3e366bc01',
      to: '0x1bc5cbf71b068642fc5e89c5e3d0bdb3e366bc02',
    },
    '17': {
      id: '15',
      type: 'sign',
      txhash: 'abcd',
      authorizedAt: 14834004807621990,
      to: '0x1bc5cbf71b068642fc5e89c5e3d0bdb3e366bc02',
    }
  }
}

const withTransactions = {
  uport: uportReducer(withIdentity.uport, storeIdentity(identityWithTransactions))
}

it('returns history', () => {
  expect(history({})).toEqual([])
  expect(history({})).toMatchSnapshot()
  expect(history(withIdentity)).toMatchSnapshot()
  expect(history(withMultipleIdentities)).toMatchSnapshot()
})

it('returns historyCount', () => {
  expect(historyCount({})).toEqual(0)
  expect(historyCount({})).toEqual(0)
  expect(historyCount(withIdentity)).toEqual(4)
  expect(historyCount(withMultipleIdentities)).toEqual(2)
})

it('returns unreadCount', () => {
  expect(unreadCount({})).toEqual(0)
  expect(unreadCount({})).toEqual(0)
  expect(unreadCount(withIdentity)).toEqual(2)
  expect(unreadCount(withMultipleIdentities)).toEqual(0)
})

it('returns unread', () => {
  expect(toJs(unread({}))).toEqual([])
  expect(toJs(unread({}))).toMatchSnapshot()
  expect(toJs(unread(withIdentity))).toMatchSnapshot()
  expect(toJs(unread(withMultipleIdentities))).toMatchSnapshot()
})

it('returns notifications', () => {
  expect(toJs(notifications({}))).toEqual([])
  expect(toJs(notifications({}))).toMatchSnapshot()
  expect(toJs(notifications(withIdentity))).toMatchSnapshot()
  expect(toJs(notifications(withMultipleIdentities))).toMatchSnapshot()
})

it('returns notificationCount', () => {
  expect(notificationCount({})).toEqual(0)
  expect(notificationCount({})).toEqual(0)
  expect(notificationCount(withIdentity)).toEqual(3)
  expect(notificationCount(withMultipleIdentities)).toEqual(0)
})

it('returns activities', () => {
  expect(toJs(activities({}))).toEqual([])
  expect(toJs(activities({}))).toMatchSnapshot()
  expect(toJs(activities(withIdentity))).toMatchSnapshot()
  expect(toJs(activities(withMultipleIdentities))).toMatchSnapshot()
})

it('returns activitiesForAddress', () => {
  expect(toJs(activitiesForAddress({}, '0x9df0e9759b17f34e9123adbe6d3f25d54b72ad6a'))).toEqual([])
  expect(toJs(activitiesForAddress({}, '34ukSmiK1oA1C5Du8aWpkjFGALoH7nsHeDX'))).toMatchSnapshot()
  expect(toJs(activitiesForAddress(withMultipleIdentities, '0x9df0e9759b17f34e9123adbe6d3f25d54b72ad6a'))).toMatchSnapshot()
  expect(toJs(activitiesForAddress(withMultipleIdentities, '34ukSmiK1oA1C5Du8aWpkjFGALoH7nsHeDX'))).toMatchSnapshot()
})

it('returns activitiesByAddress', () => {
  expect(activitiesByAddress({}, '0x9df0e9759b17f34e9123adbe6d3f25d54b72ad6a')).toEqual([])
  expect(activitiesByAddress({}, '34ukSmiK1oA1C5Du8aWpkjFGALoH7nsHeDX')).toMatchSnapshot()
  expect(activitiesByAddress(withIdentity, '0x1bc5cbf71b068642fc5e89c5e3d0bdb3e366bc01')).toMatchSnapshot()
  expect(activitiesByAddress(withIdentity, '0x1bc5cbf71b068642fc5e89c5e3d0bdb3e366bc02')).toMatchSnapshot()
  expect(activitiesByAddress(withIdentity, '0x1bc5cbf71b068642fc5e89c5e3d0bdb3e366bc03')).toMatchSnapshot()
})

it('returns unconfirmedTransactions', () => {
  expect(toJs(unconfirmedTransactions({}, '0x9df0e9759b17f34e9123adbe6d3f25d54b72ad6a'))).toEqual([])
  expect(toJs(unconfirmedTransactions({}, '34ukSmiK1oA1C5Du8aWpkjFGALoH7nsHeDX'))).toMatchSnapshot()
  expect(toJs(unconfirmedTransactions(withIdentity, '0x9df0e9759b17f34e9123adbe6d3f25d54b72ad6a'))).toMatchSnapshot()
  expect(toJs(unconfirmedTransactions(withTransactions, '34ukSmiK1oA1C5Du8aWpkjFGALoH7nsHeDX'))).toMatchSnapshot()
})
