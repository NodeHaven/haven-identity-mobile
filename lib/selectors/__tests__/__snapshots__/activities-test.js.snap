// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns activities 1`] = `Array []`;

exports[`returns activities 2`] = `
Array [
  Object {
    "attestations": Array [
      Object {
        "name": "Bob",
      },
    ],
    "authorizedAt": 14834004807621990,
    "id": "19",
    "type": "attestation",
  },
  Object {
    "attestations": Array [
      Object {
        "name": "Bob",
      },
    ],
    "authorizedAt": 14834004807621990,
    "id": "18",
    "opened": true,
    "type": "attestation",
  },
  Object {
    "id": 13,
    "iss": "0x1bc5cbf71b068642fc5e89c5e3d0bdb3e366bc02",
    "type": "attestation",
  },
  Object {
    "client_id": "0x1bc5cbf71b068642fc5e89c5e3d0bdb3e366bc03",
    "id": 12,
    "type": "disclosure",
  },
  Object {
    "authorizedAt": 14834004807621990,
    "id": 11,
    "to": "0x1bc5cbf71b068642fc5e89c5e3d0bdb3e366bc02",
    "type": "connect",
  },
  Object {
    "authorizedAt": 14834004807621990,
    "client_id": "0x1bc5cbf71b068642fc5e89c5e3d0bdb3e366bc01",
    "id": 10,
    "to": "0x1bc5cbf71b068642fc5e89c5e3d0bdb3e366bc02",
    "type": "sign",
  },
]
`;

exports[`returns activities 3`] = `
Array [
  Object {
    "authorizedAt": 14834004807621990,
    "id": "15",
    "to": "0x1bc5cbf71b068642fc5e89c5e3d0bdb3e366bc02",
    "type": "connect",
  },
  Object {
    "authorizedAt": 14834004807621990,
    "client_id": "0x1bc5cbf71b068642fc5e89c5e3d0bdb3e366bc01",
    "id": "14",
    "to": "0x1bc5cbf71b068642fc5e89c5e3d0bdb3e366bc02",
    "type": "sign",
  },
]
`;

exports[`returns activitiesByAddress 1`] = `Array []`;

exports[`returns activitiesByAddress 2`] = `
Array [
  Object {
    "authorizedAt": 14834004807621990,
    "client_id": "0x1bc5cbf71b068642fc5e89c5e3d0bdb3e366bc01",
    "id": 10,
    "to": "0x1bc5cbf71b068642fc5e89c5e3d0bdb3e366bc02",
    "type": "sign",
  },
]
`;

exports[`returns activitiesByAddress 3`] = `
Array [
  Object {
    "authorizedAt": 14834004807621990,
    "id": 11,
    "to": "0x1bc5cbf71b068642fc5e89c5e3d0bdb3e366bc02",
    "type": "connect",
  },
  Object {
    "authorizedAt": 14834004807621990,
    "client_id": "0x1bc5cbf71b068642fc5e89c5e3d0bdb3e366bc01",
    "id": 10,
    "to": "0x1bc5cbf71b068642fc5e89c5e3d0bdb3e366bc02",
    "type": "sign",
  },
]
`;

exports[`returns activitiesByAddress 4`] = `
Array [
  Object {
    "client_id": "0x1bc5cbf71b068642fc5e89c5e3d0bdb3e366bc03",
    "id": 12,
    "type": "disclosure",
  },
]
`;

exports[`returns activitiesForAddress 1`] = `Array []`;

exports[`returns activitiesForAddress 2`] = `
Array [
  Object {
    "attestations": Array [
      Object {
        "name": "Bob",
      },
    ],
    "authorizedAt": 14834004807621990,
    "id": "19",
    "type": "attestation",
  },
  Object {
    "attestations": Array [
      Object {
        "name": "Bob",
      },
    ],
    "authorizedAt": 14834004807621990,
    "id": "18",
    "opened": true,
    "type": "attestation",
  },
  Object {
    "id": 13,
    "iss": "0x1bc5cbf71b068642fc5e89c5e3d0bdb3e366bc02",
    "type": "attestation",
  },
  Object {
    "client_id": "0x1bc5cbf71b068642fc5e89c5e3d0bdb3e366bc03",
    "id": 12,
    "type": "disclosure",
  },
  Object {
    "authorizedAt": 14834004807621990,
    "id": 11,
    "to": "0x1bc5cbf71b068642fc5e89c5e3d0bdb3e366bc02",
    "type": "connect",
  },
  Object {
    "authorizedAt": 14834004807621990,
    "client_id": "0x1bc5cbf71b068642fc5e89c5e3d0bdb3e366bc01",
    "id": 10,
    "to": "0x1bc5cbf71b068642fc5e89c5e3d0bdb3e366bc02",
    "type": "sign",
  },
]
`;

exports[`returns activitiesForAddress 3`] = `
Array [
  Object {
    "authorizedAt": 14834004807621990,
    "id": "15",
    "to": "0x1bc5cbf71b068642fc5e89c5e3d0bdb3e366bc02",
    "type": "connect",
  },
  Object {
    "authorizedAt": 14834004807621990,
    "client_id": "0x1bc5cbf71b068642fc5e89c5e3d0bdb3e366bc01",
    "id": "14",
    "to": "0x1bc5cbf71b068642fc5e89c5e3d0bdb3e366bc02",
    "type": "sign",
  },
]
`;

exports[`returns history 1`] = `Array []`;

exports[`returns history 2`] = `
Array [
  Object {
    "attestations": Array [
      Object {
        "name": "Bob",
      },
    ],
    "authorizedAt": 14834004807621990,
    "id": "19",
    "type": "attestation",
  },
  Object {
    "attestations": Array [
      Object {
        "name": "Bob",
      },
    ],
    "authorizedAt": 14834004807621990,
    "id": "18",
    "opened": true,
    "type": "attestation",
  },
  Object {
    "authorizedAt": 14834004807621990,
    "id": 11,
    "to": "0x1bc5cbf71b068642fc5e89c5e3d0bdb3e366bc02",
    "type": "connect",
  },
  Object {
    "authorizedAt": 14834004807621990,
    "client_id": "0x1bc5cbf71b068642fc5e89c5e3d0bdb3e366bc01",
    "id": 10,
    "to": "0x1bc5cbf71b068642fc5e89c5e3d0bdb3e366bc02",
    "type": "sign",
  },
]
`;

exports[`returns history 3`] = `
Array [
  Object {
    "authorizedAt": 14834004807621990,
    "id": "15",
    "to": "0x1bc5cbf71b068642fc5e89c5e3d0bdb3e366bc02",
    "type": "connect",
  },
  Object {
    "authorizedAt": 14834004807621990,
    "client_id": "0x1bc5cbf71b068642fc5e89c5e3d0bdb3e366bc01",
    "id": "14",
    "to": "0x1bc5cbf71b068642fc5e89c5e3d0bdb3e366bc02",
    "type": "sign",
  },
]
`;

exports[`returns notifications 1`] = `Array []`;

exports[`returns notifications 2`] = `
Array [
  Object {
    "attestations": Array [
      Object {
        "name": "Bob",
      },
    ],
    "authorizedAt": 14834004807621990,
    "id": "19",
    "type": "attestation",
  },
  Object {
    "id": 13,
    "iss": "0x1bc5cbf71b068642fc5e89c5e3d0bdb3e366bc02",
    "type": "attestation",
  },
  Object {
    "client_id": "0x1bc5cbf71b068642fc5e89c5e3d0bdb3e366bc03",
    "id": 12,
    "type": "disclosure",
  },
]
`;

exports[`returns notifications 3`] = `Array []`;

exports[`returns unconfirmedTransactions 1`] = `Array []`;

exports[`returns unconfirmedTransactions 2`] = `Array []`;

exports[`returns unconfirmedTransactions 3`] = `
Array [
  Array [
    "15",
    "abcd",
  ],
]
`;

exports[`returns unread 1`] = `Array []`;

exports[`returns unread 2`] = `
Array [
  Object {
    "id": 13,
    "iss": "0x1bc5cbf71b068642fc5e89c5e3d0bdb3e366bc02",
    "type": "attestation",
  },
  Object {
    "client_id": "0x1bc5cbf71b068642fc5e89c5e3d0bdb3e366bc03",
    "id": 12,
    "type": "disclosure",
  },
]
`;

exports[`returns unread 3`] = `Array []`;
